build.xml

```diff
@@ -221,7 +214,15 @@
       <arg value="--always"/>
     </exec>
     <!-- Set version if git describe failed. -->
-    <property name="adaptorlib.version" value="unknown"/>
+    <property name="adaptorlib.version" value="unknown-${DSTAMP}"/>
+
+    <!-- Use the version without the "v" prefix for file names. -->
+    <loadresource property="adaptorlib.suffix">
+      <propertyresource name="adaptorlib.version"/>
+      <filterchain>
+          <replaceregex pattern="^(v(\d))?" replace="-\2"/>
```

Comment:
Yes, we always get a leading -, and strip a leading v but only if it's followed by a digit. "v321" becomes "-321", "unknown-2017" becomes "-unknown-2017", and "voodoo" becomes "-voodoo".